files/scripts/syncd.sh

```diff
@@ -136,6 +136,40 @@ wait() {
     /usr/bin/${SERVICE}.sh wait $DEV
 }
 
+collect_mst() {
+    debug "Collecting MST dump before syncd restart"
+    IRGDBSERVER=/usr/local/bin/ir-gdbserver
+    TMPDIR=/tmp/mlnxdump
+    MSTDIR=$TMPDIR/mstdump
+    DUMPDIR=/var/dump/mstdump
+    IRGDBDIR=$TMPDIR/ir-gdb
+    mkdir -p $MSTDIR
+    mkdir -p $IRGDBDIR
+    if ! mst status -v > $MSTDIR/mststatus; then
+        debug "mst status command returned error"
```

Comment:
is there apossible to print the error received? is there a clear message of the failure? it will probalby help to debug

Reply:
That's a good suggestion. The error of the above command will also be redirected to mststatus file which will be captured in the dump irrespective of it failed or succeeded